Library/Homebrew/os/mac/sdk.rb

```diff
@@ -46,9 +52,7 @@ def sdk_for(version)
 
       sig { returns(T::Array[SDK]) }
       def all_sdks
-        return @all_sdks if @all_sdks
-
-        @all_sdks = []
+        return @all_sdks unless @all_sdks.empty?
```

Comment:
Although rare, be careful as this sounds like it will be a slowdown in the case this is always empty - before it would not re-evaluate if it ran once and returned empty. I'm cautious sneaking in behaviour changes in PRs like this.